onnxruntime/core/providers/js/operators/constant_of_shape.cc

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/js/js_kernel.h"
+
+#include "constant_of_shape.h"
+
+namespace onnxruntime {
+namespace js {
+
+#define CONSTANT_OF_SHAPE_KERNEL_VERSIONED(domain, data_type, since_version, end_version) \
+  ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_EX(                                                \
+      ConstantOfShape,                                                                    \
+      domain,                                                                             \
+      since_version,                                                                      \
+      end_version,                                                                        \
+      data_type,                                                                          \
+      kJsExecutionProvider,                                                               \
+      (*KernelDefBuilder::Create())                                                       \
+          .TypeConstraint("T1", DataTypeImpl::GetTensorType<int64_t>())                   \
+          .TypeConstraint("T2", DataTypeImpl::GetTensorType<data_type>())                 \
+          .InputMemoryType(OrtMemTypeCPU, 0),                                             \
+      ConstantOfShape<data_type>);
+
+#define CONSTANT_OF_SHAPE_KERNEL(domain, data_type, since_version)        \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(                                          \
+      ConstantOfShape,                                                    \
+      domain,                                                             \
+      since_version,                                                      \
+      data_type,                                                          \
+      kJsExecutionProvider,                                               \
+      (*KernelDefBuilder::Create())                                       \
+          .TypeConstraint("T1", DataTypeImpl::GetTensorType<int64_t>())   \
+          .TypeConstraint("T2", DataTypeImpl::GetTensorType<data_type>()) \
+          .InputMemoryType(OrtMemTypeCPU, 0),                             \
+      ConstantOfShape<data_type>);
+
+CONSTANT_OF_SHAPE_KERNEL_VERSIONED(kOnnxDomain, float, 9, 19)
```

Comment:
Can we use it like [here](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/js/operators/transpose.cc#L15-L16) to support `float` and `int32_t`?

Reply:
`ConstantOfShape` is a template class here? And I need to get the data size from template parameter. If we want to use like other places to set type constraint, we must add members in `ConstantOfShapeBase` class https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/cpu/generator/constant_of_shape_base.h , I am not sure whether it is convenient for reviewing.

Reply:
I am not familiar with this part. Will leave it to other reviewers. 
Other part looks good for me. Thanks.